workflow-templates/build.yml

```diff
@@ -0,0 +1,122 @@
+        required: false
+        type: string
+      howso-validator-py-build:
+        description: |
+          (Optional) howso-validator-py build or version number. Defaults to pinned release.
+        required: false
+        type: string
+      howso-validator-enterprise-py-build:
+        description: |
+          (Optional) howso-validator-enterprise-py build or version number. Defaults to pinned release.
+        required: false
+        type: string
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      payload:
+        required: false
+        type: string
+      build-type:
+        required: false
+        type: string
+      optional-release-tag:
+        required: false
+        type: string
+
+jobs:
+
+  metadata:
+    #########################################################################
+    # METADATA JOB: Parses inputs depending on build type and sets version  #
+    # and dependency details.                                               #
+    #                                                                       #
+    # OUTPUTS:                                                              #
+    #   version: release tag, PR build version, branch build version        #
+    #   upstream-details: JSON of all dependency details, e.g.,             #
+    #     {howso-engine-py: {run-type: release, run-id: 123456, url: ... }} #
+    #########################################################################
+    uses: howsoai/.github/.github/workflows/set-metadata.yml
+    secrets: inherit
+    with:
+      # Remove unused parameters
+      payload: ${{ inputs.payload }}
+      build-type: ${{ inputs.build-type }}
+      optional-release-tag: ${{ inputs.optional-release-tag }}
+      amalgam-lang-py-build: ${{ inputs.amalgam-lang-py-build }}
+      howso-engine-py-build: ${{ inputs.howso-engine-py-build }}
+      howso-validator-py-build: ${{ inputs.howso-validator-py-build }}
+      howso-validator-enterprise-py-build: ${{ inputs.howso-validator-enterprise-py-build }}
+      howso-synthesizer-py-build: ${{ inputs.howso-synthesizer-py-build }}
+
+  #############################################################
+  # TODO: implement additional build jobs for this project    #
+  # Example Pytest and Release jobs are below.                #
+  #############################################################
+  pytest-linux-3-8-st:
+    needs: ['metadata']
+    if: inputs.build-type != 'PR'
+    uses: howsoai/.github/.github/workflows/pytest.yml
+    secrets: inherit
+    with:
+      platform: 'howso-ubuntu-runner-latest'
+      platform-pretty: 'Linux'
+      python-version: '3.8'
+      workers: '16'
+      config-fp: './config/howso-st-debug.yml'
+      config-pretty: 'Amalgam -st'
+      upstream-details: ${{ needs.metadata.outputs.upstream-details }}
+
+  release:
+    if: inputs.build-type == 'release'
+    environment:
+      name: PyPi
+    permissions:
+      contents: write
```

Comment:
This release example doesn't actually who PyPi step though, right? Might be good to add that as an example step. Or if you don't, remove the PyPi here.